### PR TITLE
Implement add and edit sites

### PR DESCRIPTION
**Pull request by jerrylin1218**
- [x] Have you read the Contributing guide?
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?
- [ ] Have you formatted / linted your code locally prior to submission?
- [ ] Have you written tests for your code changes, as applicable?

## Problem this Solves
Implements adding and editing the sites popup.

*The Zip code input could be improved quite a bit, but I left it flexible for now.

### CODE REVIEW

File: src/components/ZipCodeInput.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        onChanged(value);
+    }, [value, onChanged]);
+
+    return(
+        <div>
+            {name}<br/>
+            <input
+                name={name.split(" ").join("_").toLowerCase()}
+                type="tel"
+                pattern="[0-9-]*"
```
Comment (mwilli20) [inline src/components/ZipCodeInput.js:25]: why not something like?

```suggestion
                pattern="^(\d{5}(?:\-\d{4})?)$"
```

https://regexlib.com/Search.aspx?k=us+zip+code&c=-1&m=-1&ps=20&AspxAutoDetectCookieSupport=1

File: src/contributions/sites/SitePopup.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                        placeholder="latitude"
+                        value={lat}
+                        onChanged={setLat}
+                    />
+                    <FloatInput
+                        name="Longitude"
+                        placeholder="longitude"
+                        value={long}
+                        onChanged={setLong}
+                    />
```
Comment (mwilli20) [inline src/contributions/sites/SitePopup.js:184]: Do these not have names?

**Review by mwilli20 — COMMENTED** [review]
also totally not for right now but I wonder if the site popup actually looks the best this way... would it be better just to pick a point on a map and have all the details be auto filled from that. Like would that be easier.

File: src/contributions/sites/SitePopup.js (hunk 2 above)
Comment (jerrylin1218) [reply src/contributions/sites/SitePopup.js:184]: The "name" property will handle display of a label for these inputs.

**Comment by jerrylin1218** [comment]
> also totally not for right now but I wonder if the site popup actually looks the best this way... would it be better just to pick a point on a map and have all the details be auto filled from that. Like would that be easier.

Ooh, that'd be great - especially if volunteers could enter their data directly using a mobile phone. It could just use their GPS location.

**Pull request merged**